tests/Properties/ShortByteString.hs

```diff
@@ -0,0 +1,401 @@
+-- |
+-- Module      : Properties.ShortByteString
```

Comment:
Could we please fold this back into `Properties.ByteString` with appropriate `#ifdef BYTESTRING_SHORT` blocks? I'd rather avoid maintaining additional 400 lines of dense code.

Reply:
This will be a nightmare of ifdefs, because shortbytestring API is less than bytestring API and there may be other differences.

Reply:
I'd strongly prefer a nightmare of ifdefs. Otherwise tests will slightly drift and future me will spend a ton of time reconciling them once again. Plenty of tests have diverged already, in just 6 months since you forked them.

Reply:
How time flies...